### DISCUSSION TITLE
How can I import types

### DISCUSSION THREAD

**Discussion by StKostyantin** [wevm/wagmi#1429] [Q&A]
how can I import types for different hooks, parameters and functions?

In my case I need to import `UsePrepareContractWriteConfig`.
I can't find a way to import it...

Or how I can describe `args` for `usePrepareContractWrite`?

**Comment by tmm** [comment]
[`UsePrepareContractWriteConfig`](https://github.com/wagmi-dev/wagmi/blob/280a3d19b5b80c26dd185529cf240c275524662f/packages/react/src/index.ts#L50) is now exported (as well as some other hook types). We need to go through and export all of them at some point :)

**Reply by StKostyantin** [reply]
Oh, that's very cool!
but I still can't figure out how to describe the args for usePrepareContractWrite?

For example

```
interface SomeComponentProps {
  address: ??
  abi: ??
  functionName: ??
  args: ??
}

export const SomeComponent: FC<SomeComponentProps> = ({ address,
  abi,
  functionName,
  args,
  enabled,
  chainId}) => {
       const { config, error: error } = usePrepareContractWrite({
          address,
          abi,
          chainId,
          functionName,
          args,
          enabled,
      });
  }
```

**Reply by rodsouto** [reply]
I'm trying to do something similar using `UsePrepareContractWriteConfig`

```ts
export const useSendTx = ({address, abi, functionName, args, overrides}: UsePrepareContractWriteConfig) => {
  const [isTxSuccess, setIsTxSuccess] = useState(false)
  const [isTxError, setIsTxError] = useState(false)

  const { config } = usePrepareContractWrite({
    address,
    abi,
    functionName,
    args,
    overrides
  })
  
  const { isLoading: isWriteLoading, isError: isWriteError, data, write } = useContractWrite(config)

  const { isLoading: isTxLoading } = useWaitForTransaction({
    hash: data?.hash,
    onSuccess: (data) => {
      const isSuccess = data.status === 1
      setIsTxSuccess(isSuccess)
      setIsTxError(!isSuccess)
    },
    onError: () => {
      setIsTxSuccess(false)
      setIsTxError(true)
    }
  })

  return { isLoading: isWriteLoading || isTxLoading, isSuccess: isTxSuccess, isError: isWriteError || isTxError, write}
}
```

But the line with `useContractWrite` fails with the error

```
Types of property 'functionName' are incompatible.
Type 'string' is not assignable to type 'undefined'
```

do you know what could be the problem here?

**Reply by StKostyantin** [reply]
`export const useSendTx = <TAbi extends Abi = typeof erc20ABI, TFunctionName extends string = string>({address, abi, functionName, args, overrides}: UsePrepareContractWriteConfig<TAbi, TFunctionName>)`
